spec/celluloid/internals/logger_spec.rb

```diff
@@ -0,0 +1,137 @@
+RSpec.describe Celluloid::Internals::Logger do
+  before :each do
+    Celluloid.logger = double
+  end
+
+  describe "#debug" do
+    subject { -> { described_class.debug("debug message") } }
+    context "when logger is nil" do
+      before do
+        Celluloid.logger = nil
+      end
+      it { is_expected.not_to raise_error }
+    end
+
+    context "when logger is a Logger object" do
+      it do
+        expect(Celluloid.logger).to receive(:debug).once
+        subject.call
+      end
+
+      context "logging level is DEBUG" do
+        before do
+          Celluloid.logger = Logger.new(STDERR)
+          Celluloid.logger.level = Logger::DEBUG
+        end
+        it { is_expected.to output.to_stderr_from_any_process }
+      end
+
+      context "logging level is INFO" do
+        before do
+          Celluloid.logger = Logger.new(STDERR)
+          Celluloid.logger.level = Logger::INFO
+        end
+        it { is_expected.not_to output.to_stderr_from_any_process }
+      end
```

Comment:
@dilumn I made some tests to showcase that the level of the logger is enough to manipulate the log messages.

Reply:
@emesepadanyi  , i am not a contributor to this project, but my personal thoughts about this is that i think the reason for the global variable to exist was for performance. Since the logger is called every time now, the performance will decrease. Can we benchmark somehow what exactly is the performance difference? That would be great.

Perhaps instead of using a global variable we could use a `mattr_accessor :debug_enabled` on Celluloid and use that class variable. Which should preserve the existing performance. 

This is just an idea, might not be the best solution.

Reply:
This is a decent idea, but setting the logger to nil has almost the same effect.  The performance difference between checking a `debug_enabled` and the existing check if there is a logger set, shouldn't be great.

The one thing that does make a difference is when called the logger the string must be computed as an argument, and this can create wasted overhead if the logger isn't even going to be used.  The `Logger` class in Ruby solves for this by allowing it to be called with a block that returns a string.  The block would then only be evaluated if the logger & level would result in printing the log message.  Otherwise the block isn't evaluated.

Switching log statements that are expensive computationally, or that have interpolation, would improve performance slightly.